### ISSUE TITLE
Enforce to not use LABEL at all

### ISSUE THREAD

**Issue by abdennour** [hadolint/hadolint#353]
- [] This is a bug report
- [x] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

LABEL is not allowed

### Actual behavior
Anyone can put LABEL in any Dockerfile

# Feature Overview : 

Image labels can be set during the Build by both either "docker build" or "docker-compose.yaml". 

I want to keep consistent lables for all images. I don't want to see LABEL instruction in Dockerfile.

**Comment by SuperSandro2000** [comment]
It is not expected that you build Dockerfiles with docker-compose but with docker build.
And it would break the opencontainer spec here https://github.com/opencontainers/image-spec.

> I don't want to see LABEL instruction in Dockerfile.

Just use grep and fail CI if you find LABEL.

**Comment by abdennour** [comment]
Thank you @SuperSandro2000  !

But Think about multi-stages images where an image use another one as base. If you put labels in the first one, you will lost caching layers in the second one whenever LABELs of the first changes .. And it will be.

Just think twice before please.

```Dockerfile
FROM alpine as ssh
RUN apk add ssh-client
LABEL release-date="$DATE"

FROM ssh as ansible
RUN apk add ansible
LABEL release-date="$DATE"
```
Tell me know how you will keep the cache  of `apk add ansible`

**Comment by SuperSandro2000** [comment]
> But Think about multi-stages images where an image use another one as base. If you put labels in the first one, you will lose caching layers in the second one whenever LABELs of the first changes .. And it will.

But a reason to not use them and buildkit should fix that.

>Tell me know how you will leverage the cache of apk add ansible layer

RUN apk add ansible ssh-client

**Comment by abdennour** [comment]
Thank you @SuperSandro2000 .
I am still not totally agree . 

> RUN apk add ansible ssh-client

What about adopting the least container size principle ? 
The process A needs only ssh, the other process B needs ssh & ansible. 
Why I run the process A with a container image bigger than its need ?

**Comment by FernandoMiguel** [comment]
You can use target to produce the output at the layer you want

**Comment by abdennour** [comment]
Thank you @FernandoMiguel . However, this methodology does not comply with DRY principle. 
If the first target install "ssh", the second target should reuse the first and just add the needed without installing again "ssh".
